docs/book/src/clusterctl/provider-contract.md

```diff
@@ -150,8 +150,33 @@ will look for objects to reconcile.
 
 #### Variables
 
-The components YAML can contain environment variables matching the regexp `\${\s*([A-Z0-9_]+)\s*}`; it is highly
-recommended to prefix the variable name with the provider name e.g. `${ AWS_CREDENTIALS }`
+The components YAML can contain environment variables matching the format ${VAR}; it is highly
+recommended to prefix the variable name with the provider name e.g. `${AWS_CREDENTIALS}`
+
+<aside class="note warning">
+
+<h1>Warning</h1>
+
+`clusterctl` previously supported variables that matched the following
+formats: `${ VAR }`, `${ VAR}`,`${VAR }` and `${VAR$FOO}`.
+
+The first three formats are currently supported but will be deprecated in the
+near future.
+<br/>
+The last format is no longer supported.
```

Comment:
```suggestion
`clusterctl` supports also variables names with trailing/leading spaces (${ VAR }`, `${ VAR}`,`${VAR }`),
but this will be removed in the near future.
```
AFAIK ${VAR$FOO} is not a valid variable name; it will be also great to define" near futur, e.g. v1alpha4...

Reply:
I'll do the v1alpha4 timeline. I think that makes sense. The reason I entered the `${VAR$FOO}` was because we had tests for that specific variable (the original variable regex has a `$` in it --> `A-Za-z0-9_$`).